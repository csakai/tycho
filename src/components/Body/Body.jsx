import React from 'react';
import PropTypes from 'prop-types';
import Constants from '../../constants';
import TextureContainer from '../../containers/TextureContainer';

export default class Body extends React.Component {

  static propTypes = {
    radius: PropTypes.number.isRequired,
    rotation: PropTypes.object,
    label: PropTypes.object
  };

  componentDidMount = () => {
    this.refs.group.add(this.props.label);
  }

  render() {
    return (
      <group ref="group">
        <mesh rotation={this.props.rotation}>
          <TextureContainer textures={this.props.maps} />
          <sphereGeometry
            widthSegments={Constants.WebGL.SPHERE_SEGMENTS}
            heightSegments={Constants.WebGL.SPHERE_SEGMENTS}
            radius={this.props.radius}
          />
        </mesh>
        <axisHelper size={200} rotation={this.props.rotation} />
      </group>
    );
  }
}
