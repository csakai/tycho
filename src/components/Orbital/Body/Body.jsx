import React from 'react';
import PropTypes from 'prop-types';
import Constants from '../../../constants';
import Scale from '../../../utils/Scale';
import TextureContainer from '../../../containers/TextureContainer';
import Rings from '../../Orbital/Rings';

export default class Body extends React.Component {

    static propTypes = {
        radius: PropTypes.number.isRequired,
        rings: PropTypes.object,
        rotation: PropTypes.object,
    };

    render() {
        const {rings, radius, scale} = this.props;

        return (
            <group>
                <mesh rotation={this.props.rotation}>
                    <TextureContainer textures={this.props.maps} />
                    <sphereGeometry
                        widthSegments={Constants.WebGL.SPHERE_SEGMENTS}
                        heightSegments={Constants.WebGL.SPHERE_SEGMENTS}
                        radius={Scale(radius, scale)}
                    />
                </mesh>
                {rings && <Rings {...rings} scale={scale} />}
            </group>
        );
    }
}
