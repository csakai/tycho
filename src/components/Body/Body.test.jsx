import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {Euler} from 'three';
import Body from './';

describe('Body Component', () => {
  let component;
  const radius = 100;
  const rotation = new Euler(1, 1, 1);

  beforeEach(() => {
    component = shallow(<Body
      radius={radius}
      rotation={rotation}
    />);
  });

  it('should set the body mesh rotation to prop.rotation', () => {
    const mesh = component.find('mesh');
    const props = mesh.props();

    expect(props).toHaveProperty('rotation');
    expect(props.rotation).toEqual(rotation);
  });

  it('should set the sphereGeometry radius to prop.radius', () => {
    const sphereGeometry = component.find('sphereGeometry');
    const props = sphereGeometry.props();

    expect(props).toHaveProperty('radius');
    expect(props.radius).toEqual(radius);
  });

  it('should render the body successfully', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
