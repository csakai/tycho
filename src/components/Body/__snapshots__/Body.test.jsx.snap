// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Component should render the body successfully 1`] = `
<group>
  <mesh
    rotation={
      Euler {
        "_order": "XYZ",
        "_x": 1,
        "_y": 1,
        "_z": 1,
      }
    }
  >
    <TextureContainer />
    <sphereGeometry
      heightSegments={32}
      radius={100}
      widthSegments={32}
    />
  </mesh>
  <axisHelper
    rotation={
      Euler {
        "_order": "XYZ",
        "_x": 1,
        "_y": 1,
        "_z": 1,
      }
    }
    size={200}
  />
</group>
`;
